flocker/acceptance/testtools.py

```diff
@@ -928,6 +928,35 @@ def created(response):
     return creating
 
 
+def extract_external_port(
+    client, container_identifier, internal_port
+):
+    """
+    Inspect a running container for the external port number on which a
+    particular internal port is exposed.
+
+    :param docker.Client client: The Docker client to use to perform the
+        inspect.
```

Comment:
The tests that want to use it use docker-py and not our client.  I suppose because they don't want to have the extra complexity of our layer involved? Like, it's more code to generate failures that aren't relevant to what's being tested.  Maybe also because docker-py is synchronous and ours is asynchronous and it's more convenient to use a synchronous API than an asynchronous one.  Also ours is private inside `flocker.node`.